### PR TITLE
Extracting DB credentials from env.php

### PR DESCRIPTION
**Pull request by bertod**
Reading database dbname, username and password, like you can do in Magento 1 from local.xml

- files changed:
 1. file /magmi/web/magmi_config_setup row 190
 2. file /magmi/inc/magmi_engine from row 511 to 531 

[envscraping.zip](https://github.com/macopedia/magmi-m2/files/3906375/envscraping.zip)

### CODE REVIEW

File: magmi/inc/magmi_engine.php [hunk 1]
```diff
@@ -508,7 +508,28 @@ public function connectToMagento()
...
-            } else {
+            }else if($conn == 'envphp'){ //berto - scraping env.php
+				$baseDir = $this->getProp('MAGENTO', 'basedir');
+                $envPath = $baseDir.'/app/etc/env.php';
+                if (!file_exists($envPath)) {
+                    throw new Exception("Cannot load xml from path '$envPath'");
+                }
+				$content = rtrim(file_get_contents($envPath));
+				$db_info = array();
+				preg_match("/'host'\s*=>\s*'\w*'/",$content,$db_info['host'] );
```
Comment (tmotyl) [inline magmi/inc/magmi_engine.php]: Env.php is a php file, so cant it be just included?
Using regex for this doesnt feel right.
Also please format your code (PSR-2)

**Comment by bertod** [comment]
I rewrited it using env.php inclusion instead of using regex

**Comment by bertod** [comment]
Regarding format your code (PSR-2), I'm sorry about it, I think that is a problem from my editor, because the code isn't actually formatted bad as you can see on git. I will fix it

**Comment by fsw** [comment]
Hi @bertod ,
Thanks for contributing, 
I have merged your category attribute SQL fix as a separate merge to keep things clean. 
Please try to keep one issue per merge request.

File: magmi/inc/magmi_engine.php [hunk 2]
```diff
@@ -513,7 +513,26 @@ public function connectToMagento()
...
+				$env_array = include $envPath;
+				//$host = $GLOBALS['env_array']['db']['connection']['default']['host'];
+				$host = $env_array['db']['connection']['default']['host'];
+                //$dbname = $GLOBALS['env_array']['db']['connection']['default']['dbname'];
+                $dbname = $env_array['db']['connection']['default']['dbname'];
+                //$user = $GLOBALS['env_array']['db']['connection']['default']['username'];
+                $user = $env_array['db']['connection']['default']['username'];
+                //$pass = $GLOBALS['env_array']['db']['connection']['default']['password'];
+                $pass = $env_array['db']['connection']['default']['password'];
+                $port = '3306';
```
Comment (tmotyl) [inline magmi/inc/magmi_engine.php]: can't port also be passed as env value?

File: magmi/inc/magmi_engine.php [hunk 3]
```diff
@@ -513,7 +513,26 @@ public function connectToMagento()
                 $user = $default_setup->username;
                 $pass = $default_setup->password;
                 $port = $default_setup->port;
-            } else {
+            }else if($conn == 'envphp'){ //berto - scraping env.php
```
Comment (tmotyl) [inline magmi/inc/magmi_engine.php]: please fix code formatting and remove unnecessary comments

File: magmi/inc/magmi_engine.php (hunk 2 above)
Comment (bertod) [reply magmi/inc/magmi_engine.php]: In the env.php file I used for testing, there isn't a "port" value. So I forced the default one.

File: magmi/inc/magmi_engine.php (hunk 3 above)
Comment (bertod) [reply magmi/inc/magmi_engine.php]: Ok done. Check know please

File: magmi/inc/magmi_engine.php [hunk 4]
```diff
@@ -513,6 +513,17 @@ public function connectToMagento()
                 $user = $default_setup->username;
                 $pass = $default_setup->password;
                 $port = $default_setup->port;
+            } else if ($conn == 'envphp') { //berto - scraping env.php
```
Comment (tmotyl) [inline magmi/inc/magmi_engine.php]: Please remove this comment too, and use tripple ===

File: magmi/inc/magmi_engine.php (hunk 2 above)
Comment (tmotyl) [reply magmi/inc/magmi_engine.php]: Then please check if its set and apply default value only if its not present.

File: magmi/inc/magmi_engine.php (hunk 4 above)
Comment (bertod) [reply magmi/inc/magmi_engine.php]: done

**Comment by bertod** [comment]
@tmotyl regarding the database port, I found on the DevDocs (https://devdocs.magento.com/guides/v2.2/install-gde/install/cli/install-cli-install.html) that you can include the port in the host. So I think it will work.

**Comment by tmotyl** [comment]
please resolve the conflict, other changes looks good

**Comment by bertod** [comment]
Conflicts are fixed

**Pull request merged**